sdk/lib/core/pattern.dart

```diff
@@ -20,6 +20,20 @@ abstract class Pattern {
   /// and then from the end of the previous match (but always
   /// at least one position later than the *start* of the previous
   /// match, in case the pattern matches an empty substring).
+  /// ```dart
+  /// RegExp exp = RegExp(r"(\w+)");
+  /// String str = "Dash is a bird";
+  /// Iterable<Match> matches = exp.allMatches(str, 8);
+  /// for (Match m in matches) {
+  ///   String match = m[0]!;
+  ///   print(match);
+  /// }
+  /// ```
+  /// The output of the example is:
+  /// ```
+  /// a
+  /// bird
+  /// ```
```

Comment:
the output of example could be also inside previous code block with comments.. but this is fine.

and 
for (Match m in matches).....  Match can be final inside for loop